src/app/zones/sources.ts

```diff
@@ -0,0 +1,73 @@
+import { DataSourceResponse } from '@/app/application/services/data-source/DataSourcePool'
+import type KumaApi from '@/services/kuma-api/KumaApi'
+import type { PaginatedApiListResponse as CollectionResponse } from '@/types/api.d'
+import type { ZoneOverview, ZoneIngressOverview, ZoneEgressOverview } from '@/types/index.d'
+import { fetchAllResources } from '@/utilities/helpers'
+
+type PaginationParams = {
+  size: number
+  page: number
+}
+
+export type ZoneOverviewCollection = CollectionResponse<ZoneOverview>
+export type ZoneOverviewSource = DataSourceResponse<ZoneOverview>
+export type ZoneOverviewCollectionSource = DataSourceResponse<ZoneOverviewCollection>
+
+export type AllZoneIngressOverviewCollection = CollectionResponse<ZoneIngressOverview>
+export type AllZoneIngressOverviewSource = DataSourceResponse<ZoneIngressOverview>
+export type AllZoneIngressOverviewCollectionSource = DataSourceResponse<AllZoneIngressOverviewCollection>
+
+export type AllZoneEgressOverviewCollection = CollectionResponse<ZoneEgressOverview>
+export type AllZoneEgressOverviewSource = DataSourceResponse<ZoneEgressOverview>
+export type AllZoneEgressOverviewCollectionSource = DataSourceResponse<AllZoneEgressOverviewCollection>
+
+export type ZoneIngressOverviewCollection = CollectionResponse<ZoneIngressOverview>
+export type ZoneIngressOverviewSource = DataSourceResponse<ZoneIngressOverview>
+export type ZoneIngressOverviewCollectionSource = DataSourceResponse<ZoneIngressOverviewCollection>
+
+export type ZoneEgressOverviewCollection = CollectionResponse<ZoneEgressOverview>
+export type ZoneEgressOverviewSource = DataSourceResponse<ZoneEgressOverview>
+export type ZoneEgressOverviewCollectionSource = DataSourceResponse<ZoneEgressOverviewCollection>
+
+export const sources = (api: KumaApi) => {
+  return {
+    '/zones/zone-cps': async (params: PaginationParams, source: { close: () => void }) => {
+      source.close()
+
+      const size = params.size
+      const offset = params.size * (params.page - 1)
+
+      return await api.getAllZoneOverviews({ size, offset })
+    },
+
+    '/zones/all-zone-ingresses': async (_params: unknown, source: { close: () => void }) => {
+      source.close()
+
+      return await fetchAllResources(api.getAllZoneIngressOverviews.bind(api))
+    },
+
+    '/zones/all-zone-egresses': async (_params: unknown, source: { close: () => void }) => {
```

Comment:
Would naming this zones/zone-egresses/overviews help?

Reply:
Also why do we want this kind of calls?

Reply:
> Would naming this zones/zone-egresses/overviews help?

I chose `all-` here because of what I’m explaining below. Technically, the `/zones/zone-ingresses` and `/zones/zone-egresses` entries are also for their respective *Overview representation, but there we’re actually talking about paginated results.

> Also why do we want this kind of calls?

This looks pretty much like a hack from old times.

In short: We fetch _all_ ingresses and egresses just to be able to associated them with the Zone CPs (based on `zoneIngress.zone`/`zoneEgress.zone`) and we use this for exactly one thing: showing Yes or No in the respective column in the table. We could drop all of this if a `ZoneOverview` had something like `hasIngress: boolean` and `hasEgress: boolean`.

Reply:
Yes please drop show ZoneIngress/ZoneEgress for the moment please :) We'll reintroduce this later. (We were actually talking about showing ingress/egress counts with health instead of yes/no

Reply:
I’ve removed them in https://github.com/kumahq/kuma-gui/pull/1209/commits/23562e577553fe088f50a30d1ad8112bb3587057.